dip-ct.md

```diff
@@ -0,0 +1,166 @@
+
+### Confidential addresses (CT addresses)
+
+New Dash CTs will need a new type of address, a confidential address, to send and receive CT transactions. This means that the Dash full node will require new RPCs to create, validate, list and import confidential addresses. A prefix will need to be decided upon for these addresses so they can easily be differentiated from other Dash addresses. These addresses will be new data in wallet.dat and the code which reads and writes wallet.dat will need to also be updated to store and retrieve this data. The code which rescans blockchain history for transactions belonging to the current wallet will also need to be updated.
+
+These new CT addresses require a different base58 prefix to identify them as different from traditional Dash addresses.
+
+The exact structure of a CT address is as follows. It contains the following data:
+
+  * salt (AKA blinding factor) . 32 byte random value
+  * pk . The public key, a 33 byte secp256k1 curve point
+
+A CT address can be then generated via
+
+```
+address = base58( RIPEMD160( SHA256( salt + pk ) ) )
+```
+
+where `+` denotes concatenation. 
+
+### Confidential transactions
+
+A confidential transaction contains the following data :
+
+  * A 33 byte Pederson commitment to the amount being transferred
+  * A BP rangeproof that ensures the amount transferred is inside a certain interval between 0 and 2^N - 1
+    * To support all potential value transfers between 0 and 21M the BP rangeproof needs N equal to 52
+    * The exact size of the proof depends on the number of inputs and outputs
+  * An explicit fee, since the fee cannot be computed by the network since the amount is hidden
+  * A list of input UTXOs
+  * A list of one or more output addresses
+    * These may be normal or CT addresses
+
+
+A Pederson commitment can be thought of as a sealed box which is tamper-proof and contains a secret. Mathematically a Pederscon commitment is defined as
+
+```
+P(v,s) = v[G] + s[Q]
+```
+
+where
+
+  * P(v,s) means P is a function of the variables v and s
+  * v is a value to be committed
+  * s is a salt AKA blinding factor
+  * [G] and [Q] are elliptic curve points on secp256k1 both known to committer and verifier
+      * The committer is the creator of a transaction
+      * The verifier is any node which processes the transaction to see if it is valid
+  * x[Y] means multiplication of value x by curve point [Y]
+
+[G] and [Q] MUST be randomly chosen curve points such that
+
+```
+[Q] = d[G]
+```
+
+is unknown, which is equivalent to
+
+```
+log[G]
+```
```

Comment:
citation needed

Reply:
@PastaPastaPasta latest DIP has reference links to the original Pedersen paper and another link which uses roughly the same notation that I use above.